// Code generated by ent, DO NOT EDIT.

package ent

import (
	"context"
	"recruit/ent/disability"
	"recruit/ent/predicate"

	"entgo.io/ent/dialect/sql"
	"entgo.io/ent/dialect/sql/sqlgraph"
	"entgo.io/ent/schema/field"
)

// DisabilityDelete is the builder for deleting a Disability entity.
type DisabilityDelete struct {
	config
	hooks    []Hook
	mutation *DisabilityMutation
}

// Where appends a list predicates to the DisabilityDelete builder.
func (dd *DisabilityDelete) Where(ps ...predicate.Disability) *DisabilityDelete {
	dd.mutation.Where(ps...)
	return dd
}

// Exec executes the deletion query and returns how many vertices were deleted.
func (dd *DisabilityDelete) Exec(ctx context.Context) (int, error) {
	return withHooks(ctx, dd.sqlExec, dd.mutation, dd.hooks)
}

// ExecX is like Exec, but panics if an error occurs.
func (dd *DisabilityDelete) ExecX(ctx context.Context) int {
	n, err := dd.Exec(ctx)
	if err != nil {
		panic(err)
	}
	return n
}

func (dd *DisabilityDelete) sqlExec(ctx context.Context) (int, error) {
	_spec := sqlgraph.NewDeleteSpec(disability.Table, sqlgraph.NewFieldSpec(disability.FieldID, field.TypeInt32))
	if ps := dd.mutation.predicates; len(ps) > 0 {
		_spec.Predicate = func(selector *sql.Selector) {
			for i := range ps {
				ps[i](selector)
			}
		}
	}
	affected, err := sqlgraph.DeleteNodes(ctx, dd.driver, _spec)
	if err != nil && sqlgraph.IsConstraintError(err) {
		err = &ConstraintError{msg: err.Error(), wrap: err}
	}
	dd.mutation.done = true
	return affected, err
}

// DisabilityDeleteOne is the builder for deleting a single Disability entity.
type DisabilityDeleteOne struct {
	dd *DisabilityDelete
}

// Where appends a list predicates to the DisabilityDelete builder.
func (ddo *DisabilityDeleteOne) Where(ps ...predicate.Disability) *DisabilityDeleteOne {
	ddo.dd.mutation.Where(ps...)
	return ddo
}

// Exec executes the deletion query.
func (ddo *DisabilityDeleteOne) Exec(ctx context.Context) error {
	n, err := ddo.dd.Exec(ctx)
	switch {
	case err != nil:
		return err
	case n == 0:
		return &NotFoundError{disability.Label}
	default:
		return nil
	}
}

// ExecX is like Exec, but panics if an error occurs.
func (ddo *DisabilityDeleteOne) ExecX(ctx context.Context) {
	if err := ddo.Exec(ctx); err != nil {
		panic(err)
	}
}
